File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/screens/pages/categories/CategoriesViewModel.java
package ru.usedesk.knowledgebase_gui.screens.pages.categories;

import android.support.annotation.NonNull;

import java.util.List;

import ru.usedesk.knowledgebase_gui.screens.common.DataViewModel;
import ru.usedesk.knowledgebase_gui.screens.common.ViewModelFactory;
import ru.usedesk.knowledgebase_sdk.external.IUsedeskKnowledgeBaseSdk;
import ru.usedesk.knowledgebase_sdk.external.entity.Category;

class CategoriesViewModel extends DataViewModel<List<Category>> {

    private CategoriesViewModel(@NonNull IUsedeskKnowledgeBaseSdk usedeskKnowledgeBaseSdk, long sectionId) {
        loadData(usedeskKnowledgeBaseSdk.getCategoriesSingle(sectionId));
    }

    static class Factory extends ViewModelFactory<CategoriesViewModel> {

        private IUsedeskKnowledgeBaseSdk usedeskKnowledgeBaseSdk;
        private long sectionId;

        public Factory(@NonNull IUsedeskKnowledgeBaseSdk usedeskKnowledgeBaseSdk, long sectionId) {
            this.usedeskKnowledgeBaseSdk = usedeskKnowledgeBaseSdk;
            this.sectionId = sectionId;
        }

        @NonNull
        @Override
        protected CategoriesViewModel create() {
            return new CategoriesViewModel(usedeskKnowledgeBaseSdk, sectionId);
        }

        @NonNull
        @Override
        protected Class<CategoriesViewModel> getClassType() {
            return CategoriesViewModel.class;
        }
    }
}

File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/internal/screens/pages/categories/CategoriesViewModel.kt
package ru.usedesk.knowledgebase_gui.internal.screens.pages.categories;

import androidx.annotation.NonNull;

import java.util.List;

import ru.usedesk.knowledgebase_gui.internal.screens.common.DataViewModel;
import ru.usedesk.knowledgebase_gui.internal.screens.common.ViewModelFactory;
import ru.usedesk.knowledgebase_sdk.external.IUsedeskKnowledgeBase;
import ru.usedesk.knowledgebase_sdk.external.entity.UsedeskCategory;

class CategoriesViewModel extends DataViewModel<List<UsedeskCategory>> {

    private CategoriesViewModel(@NonNull IUsedeskKnowledgeBase usedeskKnowledgeBaseSdk, long sectionId) {
        loadData(usedeskKnowledgeBaseSdk.getCategoriesRx(sectionId));
    }

    static class Factory extends ViewModelFactory<CategoriesViewModel> {

        private IUsedeskKnowledgeBase usedeskKnowledgeBaseSdk;
        private long sectionId;

        public Factory(@NonNull IUsedeskKnowledgeBase usedeskKnowledgeBaseSdk, long sectionId) {
            this.usedeskKnowledgeBaseSdk = usedeskKnowledgeBaseSdk;
            this.sectionId = sectionId;
        }

        @NonNull
        @Override
        protected CategoriesViewModel create() {
            return new CategoriesViewModel(usedeskKnowledgeBaseSdk, sectionId);
        }

        @NonNull
        @Override
        protected Class<CategoriesViewModel> getClassType() {
            return CategoriesViewModel.class;
        }
    }
}
